File: scripts/generate_images.py
#!/usr/bin/env python3
"""Scan images/ directory and generate images.json for gallery."""
import json
import os
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
IMAGES_DIR = ROOT / "images"
OUTPUT_FILE = ROOT / "images.json"
EXTENSIONS = {".jpg", ".jpeg", ".png", ".gif", ".webp", ".svg"}


def main():
    files = []
    if IMAGES_DIR.exists():
        for name in sorted(IMAGES_DIR.iterdir(), key=lambda p: p.name):
            if name.is_file() and name.suffix.lower() in EXTENSIONS:
                files.append(name.name)
    OUTPUT_FILE.write_text(json.dumps(files, ensure_ascii=False, indent=2), "utf-8")
    print(f"Generated images.json with {len(files)} images")


if __name__ == "__main__":
    main()

File: scripts/generate-images.js
const fs = require("fs");
const path = require("path");

const ROOT = path.resolve(__dirname, "..");
const IMAGES_DIR = path.join(ROOT, "images");
const OUTPUT_FILE = path.join(ROOT, "images.json");
const EXTENSIONS = [".jpg", ".jpeg", ".png", ".gif", ".webp", ".svg"];

function main() {
  let files = [];
  if (fs.existsSync(IMAGES_DIR)) {
    files = fs
      .readdirSync(IMAGES_DIR)
      .filter((name) => {
        const ext = path.extname(name).toLowerCase();
        return EXTENSIONS.includes(ext);
      })
      .sort((a, b) => a.localeCompare(b, "zh-CN"));
  }
  fs.writeFileSync(OUTPUT_FILE, JSON.stringify(files, null, 2), "utf-8");
  console.log(`Generated images.json with ${files.length} images`);
}

main();
